extensions/game.py

```diff
@@ -39,41 +38,41 @@
 @dataclasses.dataclass
 class MafiaGameConfig:
     starting_mafia: int
-    special_roles: typing.List[players.Player]
-    ctx: commands.Context
+    special_roles: typing.List[_players.Player]
+    ctx: utils.CustomContext
     night_length: int = 90
     day_length: int = 120
 
-
 class MafiaGame:
-    def __init__(self, ctx: commands.Context, *, config: str):
+    def __init__(self, ctx: utils.CustomContext, *, config: str):
         # The discord members, we'll produce our list of players later
-        self._members: typing.List[discord.Member] = None
+        self._members: typing.Optional[typing.List[discord.Member]] = None
         # The actual players of the game
-        self.players: typing.List[players.Player] = []
+        self.players: typing.List[_players.Player] = []
 
-        self.ctx: commands.Context = ctx
+        self.ctx: utils.CustomContext = ctx
         self.is_day: bool = True
 
         # Different chats needed
         self.category = discord.CategoryChannel = None
-        self.chat: discord.TextChannel = None
-        self.info: discord.TextChannel = None
-        self.jail: discord.TextChannel = None
-        self.jail_webhook: discord.Webhook = None
-        self.mafia_chat: discord.TextChannel = None
-        self.dead_chat: discord.TextChannel = None
+        self.chat: typing.Optional[discord.TextChannel] = None
+        self.info: typing.Optional[discord.TextChannel] = None
+        self.jail: typing.Optional[discord.TextChannel] = None
+        self.jail_webhook: typing.Optional[discord.Webhook] = None
+        self.mafia_chat: typing.Optional[discord.TextChannel] = None
+        self.dead_chat: typing.Optional[discord.TextChannel] = None
 
         self._alive_game_role_name: str = "Alive Players"
-        self._alive_game_role: discord.Role = None
+        self._alive_game_role: typing.Optional[discord.Role] = None
 
         self._rand = random.SystemRandom()
-        self._config: MafiaGameConfig = None
+        self._config: typing.Optional[MafiaGameConfig] = None
         # The preconfigured option that can be provided
         self._preconfigured_config: str = config
         self._day: int = 1
         self._day_notifications = collections.defaultdict(list)
-        self._role_list: list = None
+        self._role_list: typing.Optional[list] = None
+        self._game_task: typing.Optional[asyncio.Task] = None
```

Comment:
_game_task actually doesn't exist anymore, the `stop()` method that uses it below isn't used anymore. I stop the game by cancelling the task in the `mafia.py` file now, and just calling `.cleanup()`. Should probably just remove the stop method lol